greenery/fsm.py

```diff
@@ -260,7 +266,8 @@ def concatenate(*fsms: Fsm) -> Fsm:
         """
         Concatenate arbitrarily many finite state machines together.
         """
-        alphabet = set().union(*[fsm.alphabet for fsm in fsms])
+        alphabet = fsms[0].alphabet if fsms else frozenset()
+        assert all(fsm.alphabet == alphabet for fsm in fsms), "All alphabets have to be the same"
```

Comment:
If we don't check that the alphabets are the exact same, we are going to have problems when one FSM has `{a, b}` as a symbol and other one has `{b, c}`.

This somewhat reduces the power of directly using Fsm. Maybe we could provide utilty functions to rearrange a FSM to have a different alphabet?